Add wta for help channel quoting

onerandomusername opened the pull request:
Added a wta [where to ask] command to send which channels are currently open and ready for claiming.
Must have the how-to-get-help channel added into the ids.py file as GET_HELP_CHANNEL_ID

onerandomusername committed:
Add wta for help channel quoting

Added a wta [where to ask] command to send which channels are currently open and ready for claiming.
Must have the how-to-get-help channel added into the ids.py file as GET_HELP_CHANNEL_ID

CloakerSmoker commented on cogs/ahk/ahk.py:
```diff
@@ -428,7 +429,18 @@ async def rlx(self, ctx, *, code):
+		for i, c in enumerate(channels):
+			if c.id == GET_HELP_CHANNEL_ID:
+				continue
+		channel_mentions.append(c.mention)
```
Should be inside of the loop, this will only list the final channel iterated over, including `GET_HELP_CHANNEL_ID` (if it is the final channel iterated over)

onerandomusername replied:
Thank you, fixed in https://github.com/Run1e/AceBot/pull/19.

CloakerSmoker commented on cogs/ahk/ahk.py:
```diff
@@ -428,7 +429,18 @@ async def rlx(self, ctx, *, code):
+			if c.id == GET_HELP_CHANNEL_ID:
+				continue
+		channel_mentions.append(c.mention)
+		await ctx.send("Please ask in a channel under **{}** category. Use either {} or {}.".format(open_category.name,*channel_mentions))
```
It is possible that there could be zero open channels, or just one. In those cases, this message isn't correct, so you'll want to account for `len(channel_mentions)` being less than two.

onerandomusername replied:
Thank you, fixed in https://github.com/Run1e/AceBot/pull/19.

Run1e commented:
Reopening

onerandomusername committed:
Add wta for open help channel quoting 

Added a wta [where to ask] command to send which channels are currently open and ready for claiming.
This time in the correct file.

onerandomusername committed:
Update ahk.py

Run1e commented on cogs/ahk/help.py:
```diff
@@ -495,6 +495,23 @@ async def strip_postfix(self, channel: discord.TextChannel):
+        	open_category = self.bot.get_channel(OPEN_CATEGORY_ID)
+        	channels = list(open_category.text_channels)
+        	channel_mentions = []
+        	for c in channels:
+        	    	if c.id == GET_HELP_CHAN_ID:
+                		continue
```
This is the same as c = channels[-1]

Run1e commented on cogs/ahk/help.py:
```diff
@@ -495,6 +495,23 @@ async def strip_postfix(self, channel: discord.TextChannel):
+        	if len(channel_mentions) == 0:
+            		wta_format = "No help channels are available for questions at this time. Please wait to ask a question, or ask staff for guidance."
+        	else:
+            		wta_format = "Please ask in a channel under the **{}** category. Use " + (("{}, " *  (len(channel_mentions)-1) + "or ") if len(channel_mentions)-1 > 0 else "") + "{}."
```
Not sure what is going on here

Run1e commented on cogs/ahk/help.py:
```diff
@@ -495,6 +495,23 @@ async def strip_postfix(self, channel: discord.TextChannel):
+            		wta_format = "No help channels are available for questions at this time. Please wait to ask a question, or ask staff for guidance."
+        	else:
+            		wta_format = "Please ask in a channel under the **{}** category. Use " + (("{}, " *  (len(channel_mentions)-1) + "or ") if len(channel_mentions)-1 > 0 else "") + "{}."
+        	await ctx.send(wta_format.format(open_category.name,*channel_mentions))
```
Unpacking channel_mentions here will mean it's referenced by its index in the format string, which isn't how you're accessing it, so fairly certain this format won't work

onerandomusername replied:
This does work in my tests, although if you have a better method I'm interested in trying it.

Run1e commented:
I would recommend trying to get a list of available help channels first, by doing something like
```py
help_channels = [channel for channel in open_category.help_channels if channel.id != GET_HELP_CHAN_ID]
```
and then sending the error message if that list is empty, or otherwise sending a list of channel mentions.

onerandomusername committed:
Updated wta command.

Added `ais` alias.
Added error handling if category returns None.
Refactored formatting of output.
Added a lock to prevent running while a channel is opened.
